e2e/__test__/cats.test.js

```diff
@@ -0,0 +1,185 @@
+const request = require('../request');
+const db = require('../db');
+const { signupUser } = require('../data-helpers');
+const User = require('../../lib/models/user');
+
+describe('cats api', () => {
+  beforeEach(() => db.dropCollection('cats'));
+  beforeEach(() => db.dropCollection('users'));
+
+  const testAdmin = {
+    email: 'admin@admin.com',
+    password: 'admin'
+  };
+
+  const testUser = {
+    email: 'test@test.com',
+    password: 'abc'
+  };
+  
+  const kitty = {
+    breed: 'Persian',
+    nickname: 'Smushy face'
+  };
+  
+  const kitty2 = {
+    breed: 'Angora',
+    nickname: 'Pretty'
+  };
+
+  function signinAdmin(admin = testAdmin) {
+    return request
+      .post('/api/auth/signin')
+      .send(admin)
+      .expect(200)
+      .then(({ body }) => body);
+  }
+
+  it('posts a cat', () => {
+    return signupUser(testAdmin)
+      .then(user => {
+        return User.updateById(user._id, {
+          $addToSet: {
+            roles: 'admin'
+          }
+        });
+      })
+      .then(() => {
+        return Promise.all([
```

Comment:
you only need to use promise.all when you are awaiting the return values of more than one function. Here, you can use:
```return signinAdmin()```
```.then(adminUser => {```
etc etc